### DISCUSSION TITLE
Rerouting with alternative routes

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#312] [Q&A] [answered]
When I'm planning the route from the START point to the END point, I choose one of the alternatives that Cruiser navigates. This works very well.

If I start the route not exactly from the above specified START point, Cruiser can report off the route and propose a recalculation (automatic recalculation is always disabled). This is also expected behavior.

But here is the problem: if a recalculation occurs, the alternative I originally chose is replaced by another route, the default first one, chosen by Cruiser (BRouter). Even if I choose to recalculate to the original START point. I can deal with this by pausing navigation to display all alternatives and selecting the one I am interested in again. But if the recalculation occurs somewhere in the middle of the route, will my chosen alternative be "forgotten"? If possible, it would be good if she was remembered.

_Originally posted by @Ciutnik in https://github.com/devemux86/cruiser/discussions/142#discussioncomment-7011168._

**Comment by devemux86** [comment]
@Ciutnik thanks for the report!

This shouldn't happen, so we need to fix it. :slightly_smiling_face: 

In the meantime, delete all the alternatives you don't want and keep only the desired route.
(long press on the alternative routes and "Delete")

**Reply by Ciutnik** [reply]
> In the meantime, delete all the alternatives you don't want and keep only the desired route.

Of course I did that, but it doesn't help. The "default" route is still selected, and deleted routes appear after pausing navigation.

**Reply by Ciutnik** [reply]
_It looks like the routing starts completely over again, rather than just marking the missing section. That's why I originally posted an entry in the "Smart routing" topic._

**Reply by devemux86** [reply]
Thanks, this can be reproduced, I will check it.

**Reply by devemux86** [reply]
@Ciutnik thanks for the report.

- Rerouting to the start with alternative routes
Indeed needed an improvement. Next Cruiser will use the correct alternative route.

- Rerouting to an intermediate waypoint
Rerouting is done until the selected waypoint, _the rest of the route remains unchanged_.
This partial route calculation uses the best (1st) result coming from the router.

**Reply by Ciutnik** [reply]
So this problem does not occur in the middle of the route, but only at the beginning? That's probably why no one noticed it before 🙂

**Reply by devemux86** [reply]
> So this problem does not occur in the middle of the route, but only at the beginning?

Correctly. :+1: 

> That's probably why no one noticed it before

Or maybe they don't use the alternative routes enough?

**Reply by Ciutnik** [reply]
> Or maybe they don't use the alternative routes enough?

Hmm, that might be the case. But this is an absolutely basic and well-implemented feature... at least for me 😉

Once again, it comes to mind that everyone will find something nice for themselves in Cruiser 👍

**Comment by devemux86** [comment]
@Ciutnik should be fixed in:
- https://github.com/devemux86/cruiser/discussions/329

**Reply by Ciutnik** [reply]
Great! Now it's OK. Thank you very much!

**Comment by devemux86** [comment, marked as answer]
Fixed in [Cruiser 3.0.34](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
